programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -2600,6 +2596,24 @@ void endpoint_status_change(bool woken, bool unlocked) {
 
 }
 
+void endpoint_status_change(bool woken, bool unlocked) {
+    if (woken) {
+        ZITI_LOG(INFO,"Received power resume event");
+    }
+    if (unlocked) {
+        ZITI_LOG(INFO,"Received session unlocked event");
+    }
+
+    tunnel_status_change *status_change = calloc(1, sizeof(tunnel_status_change));
+    status_change->woken = woken;
+    status_change->unlocked = unlocked;
+
+    uv_async_t *ar = calloc(1, sizeof(uv_async_t));
+    ar->data = status_change;
+    uv_async_init(main_ziti_loop, ar, endpoint_status_change_function);
```

Comment:
We still have a concurrency issue here, because `uv_async_init` modifies the loop's internal variables (appends to `async_handles`) and cannot be called from multiple threads at the same time. If this was corrected, the tunnel_status semaphore would be unnecessary because `uv_async_send` would ensure that the callback is invoked serially.

Also uv_async_init should (normally) only be called once for a given callback function; otherwise the loop's async_handles field accumulates stale async handles.

My first thought was to add a function to the tSDK that allows an application to request an arbitrary function with arbitrary args to be called on the tunneler_context's loop. But that would require the windows-service code to have a reference to the tunneler_context that was created when ziti_tunneler_init was called. So I'm thinking we either  somehow expose the tunneler_context to ServiceWorkerThread so that it can be passed back to this new "tunneler_call_function" function, or maybe its okay to assume that all tunneler contexts use the same loop.